tests/runtime/ray/test_integration.py

```diff
@@ -248,6 +248,37 @@ def test_status_after_awaiting(self, runtime: _dag.RayRuntime, trial):
                 # during start and finish of a task. Thus it is >=, not >
                 assert (status.end_time - status.start_time).total_seconds() >= 0
 
+        @pytest.mark.parametrize("trial", range(5))
+        def test_handles_ray_environment_setup_error(
+            self, runtime: _dag.RayRuntime, trial, shared_ray_conn
+        ):
+            # Given
+            wf_def = _example_wfs.cause_env_setup_error.model
+            run_id = runtime.create_workflow_run(wf_def, None, False)
+
+            # Block until wf completes
+            _wait_to_finish_wf(run_id, runtime, timeout=100)
+
+            # When
+            run = runtime.get_workflow_run_status(run_id)
+
+            # Then
+            assert (
+                run.status.state == State.FAILED
+            ), f"Invalid state. Full status: {run.status}. Task runs: {run.task_runs}"
+            # Grab the logs so we can debug when the test fails
+            logs = (
+                _ray_logs.DirectLogReader(Path(shared_ray_conn._temp_dir))
+                .get_workflow_logs(wf_run_id=run_id)
+                .env_setup
+            )
+            assert run.message in [
+                f"Could not set up runtime environment. See environment setup logs for details. `orq wf logs {run_id} --env-setup`",  # noqa: E501
+                f"The workflow encountered an issue. Please consult the logs for more information. `orq wf logs {run_id}`",  # noqa: E501
+            ], (
+                "OUT\n" + "\n".join(logs.out) + "ERR\n" + "\n".join(logs.err)
```

Comment:
**question**: is there a reason why the assertion doesn't require the `"Could not set up runtime environment."` message every time?

Reply:
Running locally, there's nothing in the captured logs that serves as a reliable flag for 'env setup went wrong'. Running in the CI, RuntimeEnvSetupError is captured. Therefore this test will fail locally if we only expect `"Could not set up runtime environment."` This can be removed if [ORQSDK_978](https://zapatacomputing.atlassian.net/browse/ORQSDK-978?atlOrigin=eyJpIjoiNTA0NDQ2NzMwZjc0NDlkYmFkOGU0OTY5NmQzYzMxNDQiLCJwIjoiaiJ9) manages to unify the behaviour. I've added a todo to the test with this context.